google/cloud/storage/tests/bucket_integration_test.cc

```diff
@@ -113,6 +113,28 @@ TEST_F(BucketIntegrationTest, BasicCRUD) {
   BucketMetadata updated_meta = client.UpdateBucket(bucket_name, update);
   EXPECT_EQ(desired_storage_class, updated_meta.storage_class());
 
+  // Patch the metadata to change the storage class, add some lifecycle
+  // rules, and the website settings.
+  BucketMetadata desired_state = updated_meta;
+  LifecycleRule rule(LifecycleRule::ConditionConjunction(
+                         LifecycleRule::MaxAge(30),
+                         LifecycleRule::MatchesStorageClassStandard()),
+                     LifecycleRule::Delete());
+  desired_state.set_storage_class(storage_class::Standard())
+      .set_lifecycle(BucketLifecycle{{rule}})
+      .set_website(BucketWebsite{"index.html", "404.html"});
+
+  BucketMetadata patched =
+      client.PatchBucket(bucket_name, updated_meta, desired_state);
+  EXPECT_EQ(storage_class::Standard(), patched.storage_class());
```

Comment:
Seems like we ought to have: 
1) A test like this that only sets some fields
2) A more exhaustive test that alters all the fields and makes sure the resulting bucket has all the correct attributes (i.e. that the PATCH body was formed correctly and the given request acts as expected).

Thoughts?

Reply:
I created a larger test, which found some bugs :tada: